### DISCUSSION TITLE
Tap maintenance and Homebrew development

### DISCUSSION THREAD

**Discussion by vovictort** [Homebrew/discussions#5443] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
> brew config
HOMEBREW_VERSION: 4.3.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3f08c75e7b950d4340dab462f3e7f77e8093fa2b
Last commit: 4 days ago
Core tap JSON: 13 Jun 22:20 UTC
Core cask tap JSON: 13 Jun 22:20 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 14
Homebrew Ruby: 3.3.2 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.2/bin/ruby
CPU: 14-core 64-bit arm_palma
Clang: 15.0.0 build 1500
Git: 2.45.2 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1.1-arm64
CLT: 15.0.0.0.1.1694021235
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
> brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 15.4.


Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  node@16
  python-charset-normalizer
  python-idna
  python-requests
  python-urllib3
  terraform
```

### Description of issue

A couple of days ago, I ran `brew upgrade`, and it bumped up the major version of `ansible` from `ansible@9` to `ansible@10`. The servers I am working on are RHEL Linux, so this is a breaking change for me. The servers I am working on have Python 3.6.8 or even Python 2.7. We're just starting to use Ansible to manage servers, and once we got to a critical mass or services managed, we would migrate to new servers with updated OSs and software versions.

I have a workaround of making a devcontainer to install an older version of Ansible, but it's a pain because I don't have the same workflow availability in the devcontainer as I have configured on my machine.

Who do I contact to have an `ansible@9` installation target available in `brew`?

**Comment by SMillerDev** [comment]
> Who do I contact to have an ansible@9 installation target available in brew?

Nobody, but you can use `brew extract` to maintain your own copies of older formulae

**Comment by daeho-ro** [comment, marked as answer]
Opened a PR for ansible@9 but if this version change is really critical for you, then it is better to create your own tap.
FYI, https://github.com/Homebrew/homebrew-core/pull/174671

**Reply by daeho-ro** [reply]
The PR is merged and you can now use `ansible@9`

**Comment by vovictort** [comment]
Thank you @daeho-ro 

I will need to spend some time learning how to make a tap, and replicating the work you did here to be more self-sufficient
